### PR TITLE
docs:  add `dangerouslySetInnerHTML`

### LINKED ISSUE #1675: [DOCS] Add `dangerouslySetInnerHTML` with Qwik to the docs
**How can the documentation be improved?**

This is an issue for our docs hackathon

### LINKED ISSUE #2403: [📖] add documentation for `<RenderOnce>`
### Suggestion

noticed that there isn't a doc on it after reading #1883

### PR DESCRIPTION
**Pull request by zanettin**
fixes #1675 

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by zanettin** [comment]
@shairez @manucorporat: didn't find a good example to demonstrate `<RenderOnce>` on the docs. do you have smth i could use here?

File: packages/docs/src/routes/docs/components/rendering/index.mdx [hunk 1]
```diff
@@ -93,6 +93,13 @@ export const Parent = component$(() => {
 
 > note: it is not recommended to use the array's index as the key unless you can guarantee that the data for a given key will always be the same. It is always preferred to use some unique identifier from the data as the key.
 
+### `dangerouslySetInnerHTML`
+Qwik offers an attribute on HTML Elements called `dangerouslySetInnerHTML` as replacement for calling `innerHTML` on the DOM. Due to cross-site-scripting (XSS) possibilities by rendering not trustworthy content, you must use this attribute to remind yourself that this opperation might be dangerous.
```
Comment (literalpie) [inline packages/docs/src/routes/docs/components/rendering/index.mdx:97]:
```suggestion
Qwik offers an attribute on HTML Elements called `dangerouslySetInnerHTML` as a replacement for calling `innerHTML` on the DOM. Due to cross-site-scripting (XSS) possibilities by rendering not trustworthy content, you must use this attribute to remind yourself that this opperation might be dangerous.
```

**Comment by zanettin** [comment]
thanks @literalpie 🙏

File: packages/docs/src/routes/docs/components/rendering/index.mdx [hunk 2]
```diff
@@ -102,3 +109,22 @@ Most current-generation frameworks have a synchronous `render()` process. Synchr
 ## DOM update buffering
 
 The asynchronous nature of `render()` means that users may see an intermediate rendering of the UI as components download. Seeing an intermediate state is undesirable; therefore, Qwik will buffer all DOM updates and only flush the DOM operations once all of the components have been downloaded and their JSX functions executed. The result is that the UI will update as an atomic operation, and the user will not see the intermediate steps.
+
+## `<RenderOnce>`
+
+Qwik provides a Component called `<RenderOnce>`. It prevents re-renderings of nested children. Values which are connected to a state will always get updated.
```
Comment (shairez) [inline packages/docs/src/routes/docs/components/rendering/index.mdx:115]: I think this is missing a use case
@manucorporat should this API be documented? and if so, is there a good use case for using this in apps?

Comment (zanettin) [reply packages/docs/src/routes/docs/components/rendering/index.mdx:115]: can drop that part to get the rest merged 👍 @shairez

Comment (shairez) [reply packages/docs/src/routes/docs/components/rendering/index.mdx:115]: thanks @zanettin 
if you can, please open a separate PR for it and consider adding the comment I've written asking for a use case

I think it's valuable to know when we should consider using this so it won't confuse developers

and again, thanks for your contribution(s)! ❤

Comment (zanettin) [reply packages/docs/src/routes/docs/components/rendering/index.mdx]: @shairez done 🎉 🙏  new PR is here https://github.com/BuilderIO/qwik/pull/2848

**Comment by shairez** [comment]
awesome job @zanettin !
merging!

**Pull request merged**